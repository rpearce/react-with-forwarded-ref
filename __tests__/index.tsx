import React, { FC, ReactNode, RefObject, createRef } from 'react'
import { render } from '@testing-library/react'
import withForwardedRef from '../source'

interface Props<A = any> {
  children: ReactNode
  className: string
  forwardedRef?: RefObject<A>
}

describe('withForwardedRef', () => {
  it('renders', () => {
    const Comp: FC<Props> = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()
    const { asFragment } = render(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('renders, custom displayName', () => {
    const Comp: FC<Props> = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    Comp.displayName = 'FooBar'
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()
    const { asFragment } = render(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('ref points to HTMLDivElement', () => {
    const Comp: FC<Props> = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()

    render(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )

    expect(ref.current).toBeInstanceOf(HTMLDivElement)
  })
})
