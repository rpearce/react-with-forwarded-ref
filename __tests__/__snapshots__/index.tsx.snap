// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withForwardedRef renders 1`] = `
<DocumentFragment>
  <div
    class="foo"
  >
    Testing 123
  </div>
</DocumentFragment>
`;

exports[`withForwardedRef renders, custom displayName 1`] = `
<DocumentFragment>
  <div
    class="foo"
  >
    Testing 123
  </div>
</DocumentFragment>
`;
